### PR TITLE
doc: update file keywords v1

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Update file.magic and file.name keyword information to reflect recent updates
- Update libmagic dependency information and reference
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by jufajardini** [comment]
Related to: https://redmine.openinfosecfoundation.org/issues/6194
but not sure if we need a new ticket for the doc update in this case.

**Review by jufajardini — COMMENTED** [review]
Looks good to me, but as this may be the solution requested by @catenacyber in https://redmine.openinfosecfoundation.org/issues/437, let's wait for his feedback, too :)

File: doc/userguide/rules/file-keywords.rst [hunk 1]
```diff
@@ -5,21 +5,25 @@ Suricata comes with several rule keywords to match on various file
...
+  file.name:<string>;
 
 Example::
 
-  filename:"secret";
+  file.name:"secret";
 
 ``file.name`` supports multiple buffer matching, see :doc:`multi-buffer-matching`.
 
+**Note** The ``filename`` keyword is still supported but the
```
Comment (victorjulien) [inline doc/userguide/rules/file-keywords.rst:23]: this is no longer true:

`filename:"secret";` is now equivalent to `file.name; content:"secret"; nocase;`

Similarly, fileext:
`fileext:pdf;` is equivalent to `file.name; content:".pdf"; nocase; endswith;`. Note the dot that is prepended to `pdf`.

This is handled at parsing, so at runtime this is identical in behavior and performance.

btw we're looking at backporting this optimization to 6: https://redmine.openinfosecfoundation.org/issues/6203

File: doc/userguide/rules/file-keywords.rst [hunk 2]
```diff
@@ -33,24 +37,33 @@ Example::
...
----------
+file.magic
+----------
 
 Matches on the information libmagic returns about a file.
 
 Syntax::
 
-  filemagic:<string>;
+  file.magic:<string>;
```
Comment (victorjulien) [inline doc/userguide/rules/file-keywords.rst:47]: This incorrect. `filemagic` is the legacy notation, that uses:
`filemagic:"Windows"`;
`file.magic` is the new implementation:
`file.magic; content:"Windows"; nocase;`

Note that like with `filename` above, the difference exists only during parsing in 7.

File: doc/userguide/rules/file-keywords.rst [hunk 3]
```diff
@@ -33,24 +37,33 @@ Example::
...
+version/implementation of libmagic. Different versions and
+implementations of libmagic do not return the same information.
+Additionally there are varying Suricata performance impacts
+based on the version and implementation of libmagic.
+Additional information about Suricata and libmagic can be found
+here: https://redmine.openinfosecfoundation.org/issues/437
 
 ``file.magic`` supports multiple buffer matching, see :doc:`multi-buffer-matching`.
 
+**Note** The ``filemagic`` keyword is still supported but the
```
Comment (victorjulien) [inline doc/userguide/rules/file-keywords.rst:63]: perf should now be identical

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline comments

**Comment by jmtaylor90** [comment]
continued in #9316

**Pull request closed without merging**